cartographer/sensor/point_cloud.h

```diff
@@ -42,6 +45,7 @@ using TimedPointCloud = std::vector<TimedRangefinderPoint>;
 struct PointCloudWithIntensities {
   TimedPointCloud points;
   std::vector<float> intensities;
+  std::vector<Uint8Color> colors;
```

Comment:
With the change, the struct name does not fit to the struct anymore. 
I would expect this to have side effects for the default behavior in Cartographer when we do not need colors.
Please create a separate type e.g. PointCloudWithIntensitiesColors.

Reply:
I generally agree with this. 
In this case cartographer only has this single struct and no struct without intensities, even though intensities are optional. To avoid a lot of changes in cartographer and to be congruent with the implementation of intensities colors was added as optional attribute.
When no colors exist, the colors vector is just empty.